### DISCUSSION TITLE
Should we run the Precompiles tests once weekly?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#582] [Ideas]
One of the most common reasons for git conflicts and CI failure nowadays are the Precompiles tests failing:

https://github.com/sablier-labs/v2-core/blob/36223aeb1867b2fc197978fca19cbd5fcc538914/test/utils/Precompiles.t.sol

@andreivladbrg, what would you say about refactoring the CI workflows to only run the Precompiles tests once weekly, just like the CI Deep workflow?

**Comment by PaulRBerg** [comment]
any thoughts about this, @andreivladbrg?

**Reply by andreivladbrg** [reply]
it sounds good, but it solves only half of the problem, git conflicts would remain, which imo are the bigger problem here

**Reply by PaulRBerg** [reply]
good point

**Comment by PaulRBerg** [comment]
Any additional thoughts on this, @andreivladbrg, @smol-ninja?

I think that disabling the utils CI checks would minimize the error rates in PRs.

Maybe we could incorporate the utils tests in the [daily cron schedule](https://github.com/sablier-labs/v2-core/issues/790) for fork tests.

**Reply by andreivladbrg** [reply]
> I think that disabling the utils CI checks would minimize the error rates in PR

not sure what you mean

the only thing would be that the `CI` would not fail on each PR

> Maybe we could incorporate the utils tests in the https://github.com/sablier-labs/v2-core/issues/790 for fork tests.

as said above, agree, it should run only on `main`

**Reply by PaulRBerg** [reply]
> not sure what you mean

Remove the [`test-utils`](https://github.com/sablier-labs/v2-core/blob/7819a3b5122036596edff3326b97a8c03750bbb8/.github/workflows/ci.yml#L39) job from the `CI` workflow so that it is not run on every push/ PR.

> the only thing would be that the CI would not fail on each PR

Exactly

> as said above, agree, it should run only on `main`

I propose to run it only once every ~24 hours or so. Not on every push to `main`.

**Reply by andreivladbrg** [reply]
> so that it is not run on every push/ PR

yes, but don't see how is it related to this "minimize the error rates in PRs"

> I propose to run it only once every ~24 hours or so. Not on every push to main

yes, periodically every 24 hours, but on branch `main`, not for example on `staging`

**Reply by PaulRBerg** [reply]
> but don't see how is it related to this "minimize the error rates in PRs"

The `test-utils` job will not run anymore, leading to fewer failed jobs in CI PR reviews.

We will prevent this:

<img width="400" alt="Screenshot 2024-01-22 at 2 57 44 PM" src="https://github.com/sablier-labs/v2-core/assets/8782666/e914b0e6-c373-4f94-9cf0-1d6337e487ab">

> periodically every 24 hours, but on branch main, not for example on staging

Yes.

**Reply by andreivladbrg** [reply]
> We will prevent this:

yeah, but this is not an "error", nvm, wording game

we agree

**Reply by PaulRBerg** [reply]
> wording game

Yes. "error", "failure", whatever.

**Reply by smol-ninja** [reply]
I am also in favour of removing the `test-utils` job from every PR and only running on the `main` branch periodically. 

And what about fork tests? For example, [when a PR in v2-periphery depends on new changes in v2-core](https://github.com/sablier-labs/v2-periphery/pull/264), the fork test will always fail until we deploy the v2-core on the mainnet. So I agree with @andreivladbrg that it should also run only on the main.

**Comment by smol-ninja** [comment]
Since we all have agreed to move precompile tests alongside the fork tests schedule, should I close this discussion and create an issue out of it?

**Reply by PaulRBerg** [reply]
Yes

**Comment by smol-ninja** [comment]
Closing it.
- https://github.com/sablier-labs/v2-core/issues/832
- https://github.com/sablier-labs/v2-periphery/issues/296

**Comment by smol-ninja** [comment]
I would suggest that we should bring the fork tests back into ci with less number of runs (~ 20). During PR reviews, it would be helpful to see fork tests passing with the new changes. What's your opinions?

**Reply by PaulRBerg** [reply]
Agree
